### DISCUSSION TITLE
No completions in REPL

### DISCUSSION THREAD

**Discussion by kresdjan** [joaotavora/sly#565] [Q&A]
I don't get any completions in the REPL. I `Compile and LoadFile` `C-c C-k`.  I can't complete any function definitions or even the built-ins like `defun`. I just see a blank `*sly-completions*` buffer, the warning/error:  `Wrong type argument: stringp, #<buffer *sly-completions*>`
 
I don't think it is Sly bug but properly a misconfiguration on my part.
Does anyone have a fix, I've just switched from SLIME I've never had this problem.

The Sly configuration I've made so far is: 
```
(setq inferior-lisp-program "sbcl")
(setq sly-complete-symbol-function 'sly-flex-completions)
```

**Comment by kresdjan** [comment]
Seems that `sly-symbol-completion-mode` was to be set to `nil`.

```
(setq sly-symbol-completion-mode nil)
```

**Comment by joaotavora** [comment]
Can you tell me which Emacs version you are using.  Doing this setting `(setq sly-symbol-completion-mode nil)`  shouldn't in principle be necessary.  I think you may be seeing a bug with Emacs 29 (which I have yet to report).

**Reply by kresdjan** [reply]
I'm on the latest MacPorts dev version I think.
`GNU Emacs 30.0.50 (build 1, x86_64-apple-darwin22.2.0, NS appkit-2299.30 Version 13.1 (Build 22C65)) of 2023-01-03`

**Comment by joaotavora** [comment]
Can you tell me which Emacs version you are using.  Doing this setting `(setq sly-symbol-completion-mode nil)`  shouldn't in principle be necessary.  I think you may be seeing a bug with Emacs 29 (which I have yet to report).

**Reply by kresdjan** [reply]
Emacs 30.0.50
